hs_composite_resource/models.py

```diff
@@ -0,0 +1,180 @@
+import os
+
+from mezzanine.pages.page_processors import processor_for
+
+from hs_core.models import BaseResource, ResourceManager, resource_processor
+
+from hs_file_types.models import GenericLogicalFile
+
+
+class CompositeResource(BaseResource):
+    objects = ResourceManager("CompositeResource")
+
+    class Meta:
+        verbose_name = 'Composite Resource'
+        proxy = True
+
+    @property
+    def can_be_public_or_discoverable(self):
+        # resource level metadata check
+        if not super(CompositeResource, self).can_be_public_or_discoverable:
+            return False
+
+        # filetype level metadata check
+        for lf in self.logical_files:
+            if not lf.metadata.has_all_required_elements():
+                return False
+
+        return True
+
+    def set_default_logical_file(self):
+        """sets an instance of GenericLogicalFile to any resource file objects of this instance
+        of the resource that is not already associated with a logical file. """
+
+        for res_file in self.files.all():
+            if not res_file.has_logical_file:
+                logical_file = GenericLogicalFile.create()
+                res_file.logical_file_content_object = logical_file
+                res_file.save()
+
+    @property
+    def supports_logical_file(self):
+        """ if this resource allows associating resource file objects with logical file"""
+        return True
+
+    def get_metadata_xml(self, pretty_print=True):
+        from lxml import etree
+
+        # get resource level core metadata as xml string
+        xml_string = super(CompositeResource, self).get_metadata_xml(pretty_print=False)
+        # add file type metadata xml
+
+        # create an etree xml object
+        RDF_ROOT = etree.fromstring(xml_string)
+
+        # get root 'Description' element that contains all other elements
+        container = RDF_ROOT.find('rdf:Description', namespaces=self.metadata.NAMESPACES)
+
+        for lf in self.logical_files:
+            lf.metadata.add_to_xml_container(container)
+
+        return etree.tostring(RDF_ROOT, pretty_print=pretty_print)
+
+    def supports_folder_creation(self, folder_full_path):
```

Comment:
This is very much over-complicated. All one has to check is that the folder path is underneath the file path, and that the resource generally supports folders. I have nicer ways of checking this forthcoming.